pkg/csi_driver/multishare_ops_manager.go

```diff
@@ -337,17 +337,71 @@ func (m *MultishareOpsManager) instanceNeedsExpand(ctx context.Context, share *f
 		sumShareBytes = sumShareBytes + s.CapacityBytes
 	}
 
-	var remainingBytes int64
-	remainingBytes = share.Parent.CapacityBytes - sumShareBytes
-	if remainingBytes < share.CapacityBytes {
-		alignBytes := util.AlignBytes(share.CapacityBytes+sumShareBytes, util.GbToBytes(share.Parent.CapacityStepSizeGb))
+	remainingBytes := share.Parent.CapacityBytes - sumShareBytes
+	if remainingBytes < capacityNeeded {
+		alignBytes := util.AlignBytes(capacityNeeded+sumShareBytes, util.GbToBytes(share.Parent.CapacityStepSizeGb))
 		targetBytes := util.Min(alignBytes, util.MaxMultishareInstanceSizeBytes)
 		return true, targetBytes, nil
 	}
 	return false, 0, nil
 }
 
-func (m *MultishareOpsManager) checkAndStartShareDeleteWorkflow(ctx context.Context, instanceScPrefix string, share *file.Share) (*Workflow, error) {
+func (m *MultishareOpsManager) expandInstanceAndStartWorkflow(ctx context.Context, share *file.Share, reqBytes int64) (*Workflow, error) {
+	m.Lock()
+	defer m.Unlock()
+
+	ops, err := m.listMultishareResourceRunningOps(ctx)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	expandShareOp, err := containsOpWithShareTarget(share, util.ShareUpdate, ops)
+	if err != nil {
+		return nil, err
+	}
+	if expandShareOp != nil {
+		return &Workflow{share: share, opName: expandShareOp.Id, opType: expandShareOp.Type}, nil
+	}
+
+	// no existing share Expansion, proceed to instance check
+	err = m.verifyNoRunningInstanceOrShareOpsForInstance(share.Parent, ops)
+	if err != nil {
+		klog.Infof("Instance %v has running Op.", share.Parent.Name)
```

Comment:
this log does not provide much info.  If needed we can print the error string which contains the running op details.

Reply:
done